app/assets/javascripts/Components/peer_reviews_manager.jsx

```diff
@@ -0,0 +1,334 @@
+    ];
+  };
+
+  reviewerInRevieweeChange = (event) => {
+    const { onReviewerChangeInRevieweeTable } = this.props;
+    const isChecked = event.currentTarget.checked;
+    const reviewerGroupId = parseInt(event.currentTarget.getAttribute('data-reviewer-group-id'), 10);
+    const revieweeGroupId = parseInt(event.currentTarget.getAttribute('data-reviewee-group-id'), 10);
+    onReviewerChangeInRevieweeTable(reviewerGroupId, revieweeGroupId, isChecked);
+  };
+
+  checkboxShouldBeChecked = (reviewee_group_id, reviewer_group_id) => {
+    let returnVal = false;
+    if (reviewee_group_id in this.props.selectedReviewerInRevieweeGroups) {
+      returnVal = reviewer_group_id in this.props.selectedReviewerInRevieweeGroups[reviewee_group_id];
+    }
+    return returnVal;
+  };
+
+  render() {
+    const groups_data = this.props.groups.map(function(group) {
+      let reviewerGroups = [];
+      const reviewee_group_id = group.id;
+      const reviewer_ids = this.props.revieweeToReviewers[reviewee_group_id];
+      reviewer_ids.forEach(function(reviewer_group_id) {
+        const reviewer_group_name = this.props.groupIdToName[reviewer_group_id];
+        reviewerGroups.push(<div key={reviewer_group_id}>
+          <input id={reviewer_group_id}
+                 type='checkbox'
+                 data-reviewer-group-id={reviewer_group_id}
+                 data-reviewee-group-id={reviewee_group_id}
+                 checked={this.checkboxShouldBeChecked(reviewee_group_id, reviewer_group_id)}
+                 onChange={evt => this.reviewerInRevieweeChange(evt)}
+          /> {reviewer_group_name}</div>);
+      }.bind(this));
+
+      return {
+        _id: group.id,
+        name: group.name,
+        members: reviewerGroups,
+        section: group.section,
+        count: reviewer_ids.length
+      };
+    }.bind(this));
+
+    return (
+      <CheckboxTable
+        ref={(r) => this.checkboxTable = r}
+
+        data={groups_data}
+        columns={this.getColumns()}
+        defaultSorted={[
+          {
+            id: 'name'
+          }
+        ]}
+        loading={this.props.loading}
+        filterable
+
+        {...this.props.getCheckboxProps()}
+      />
+    );
+  }
+}
+
+
+const ReviewersTable = withSelection(RawReviewersTable);
+const RevieweesTable = withSelection(RawRevieweesTable);
+
+
+class GradersActionBox extends React.Component {
+  render = () => {
+    const { performAction } = this.props;
+
+    return (
+      <div style={{ display: 'flex' }}>
+        <div className="peer-review-amount-spinner" style={{ width: '100%' }}>
+          <span>Number of reviews per group: </span>
+          <input type="number" id="peer-review-spinner" min={1} defaultValue={1}
+                 onChange={evt => this.props.updateNumReviewers(evt.target.value)} />
+        </div>
+        <div className='rt-action-box icon'>
+          <button
+            className='assign-all-button'
+            onClick={performAction.bind(this, 'assign')}
```

Comment:
The `bind` here shouldn't be necessary. If you want to delay evaluation, you can do `() => performAction('assign')`.